src/lib/npm.js

```diff
@@ -71,16 +74,16 @@ async function validateToken(otp, uri, body, npm) {
     return false;
   }
 
-  return new Promise(resolve => {
-    client.request(uri, {method: 'PUT', auth: {otp}, body}, (err, parsed, raw, response) => {
-      if (err || !response.body.ok) {
-        resolve('Invalid authentication code');
-      } else {
-        npm.token = response.body.token;
-        resolve(true);
-      }
-    });
-  });
+  try {
+    const response = await pify(client.request.bind(client))(uri, {method: 'PUT', auth: {otp}, body});
+    if (response && response.ok) {
+      npm.token = response.token;
+      return true;
+    }
+  } catch (err) {
+    // Invalid 2FA token
```

Comment:
We should check if we got a 'E401' again. In other cases this error should get logged.

Reply:
What would be the point of checking for `E401`? How would the handling of the `E401` case would be different?
If we get a `E401` here then the token is invalid, so we return `'Invalid authentication code'` and the question get asked again.
I don't understand what you mean by "since the error may not be fixed by sending a new otp".

Reply:
Yes, if we receive status code `401` the token is invalid and everything should be handled as it is now. But if it is any other `4xx` or `5xx` status code, that error can not necessarily be fixed by entering another authentication code. So in that case the error should be logged and the function should return `true`, so that we display the message `Could not login to npm` and abort the login.

Reply:
Ok, make sense. However this PR is about fixing the login error and clean the promisification. I'd like to limit the scope, for now and merge it to solve the issue. In the current version we don't limit the errors we catch to `401` so I'd rather keep it that way in this PR.

Feel free to open another PR to improve that.

Btw the cli repo is really not well maintained (we don't even have unit tests) and there is a lot of issues to investigate and fix.
We are also open to a rewrite of some part of the code, even the entire app if we find someone committed to complete the rewrite.

Would you be interested in working on the cli project?

Reply:
See #201